### PR TITLE
Access to "sub" index (when OID has multiple indexes)

### PR DESCRIPTION
**Pull request by PipoCanaja**
Following a discussion that took place a couple of weeks ago, here is a proposal to give access in yaml to all the parts of an index. 

Simple example with the FANs in Huawei VRP devices:

```
        state:
             data:
                -
                     oid: hwFanStatusTable
                     value: hwEntityFanState
                     num_oid: '.1.3.6.1.4.1.2011.5.25.31.1.1.10.1.7.{{ $index }}'
                     descr: 'Slot {{ $subindex0 }} Fan {{ $subindex1 }} Status'
                     group: Fans
                     state_name: hwEntityFanState
                     states:
                         - { descr: Normal, graph: 1, value: 1, generic: 0 }
                         - { descr: Abnormal, graph: 1, value: 2, generic: 2 }
```
This OID is indexed by "slot" and by "fanID". So in a stack of switches, you will have : 

hwEntityFanState.<stackMemberId>.<fanId> 

So, for the 2nd fan of the 1st stack member,  instead of : 
  "descr: 'Fan {{ $subindex}} Status'" which would show "Fan 1.2 Status"
you can do : 
  "descr: 'Slot {{ $subindex0 }} Fan {{ $subindex1 }} Status'" which shows "Slot 1 Fan 2 Status"

The variables are named $subindex<X> with X from 0 to n, parsing left to right the OID exploded around ".". 

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by PipoCanaja** [comment]
OK, ready for review. Let me know if you are OK with the implementation, naming (we could also use {{ $index[0] }} to look a little bit more "phpish" :) )

File: LibreNMS/Device/YamlDiscovery.php [hunk 1]
```diff
@@ -111,7 +111,13 @@ public static function replaceValues($name, $index, $count, $data, $pre_cache)
         $value = dynamic_discovery_get_value($name, $index, $data, $pre_cache);
         if (is_null($value)) {
             // built in replacements
+            // prepare the $subindexX match variable replacement
+            $subindexes = explode('.', $index);
+            foreach (array_keys($subindexes) as $pos) {
```
Comment (murrant) [inline LibreNMS/Device/YamlDiscovery.php]: You didn't initialize $subindexStr, you could probably make this a little more readable.

Comment (PipoCanaja) [reply LibreNMS/Device/YamlDiscovery.php]: done

**Comment by murrant** [comment]
I feel like that is a little easier to follow.  Look ok to you?

**Comment by PipoCanaja** [comment]
Ooooh ;) Nice, agreed !

**Comment by PipoCanaja** [comment]
Interesting. Travis failed on a file that is not part of this PR. Any clue how that can happen ?

**Comment by murrant** [comment]
Yeah, the PR for that had style issues, but skipped checking them because the code to make testing faster was incorrect.

**Pull request merged**